src/main/java/org/folio/rest/service/action/BulkActionService.java

```diff
@@ -110,18 +127,22 @@ protected Feefineaction createFeeFineActionAndUpdateAccount(Account account, Mon
     final Feefineaction feeFineAction = new Feefineaction()
       .withAmountAction(amount.toDouble())
       .withComments(request.getComments())
-      .withNotify(request.getNotifyPatron())
-      .withTransactionInformation(((DefaultBulkActionRequest) request).getTransactionInfo())
       .withCreatedAt(request.getServicePointId())
       .withSource(request.getUserName())
-      .withPaymentMethod(defaultRequest.getPaymentMethod())
       .withAccountId(account.getId())
       .withUserId(account.getUserId())
       .withBalance(remainingAmountAfterAction.toDouble())
       .withTypeAction(actionType)
       .withId(UUID.randomUUID().toString())
       .withDateAction(new Date());
 
+    if (request instanceof DefaultBulkActionRequest) {
+      DefaultBulkActionRequest defaultBulkActionRequest = (DefaultBulkActionRequest) request;
+      feeFineAction.setPaymentMethod(defaultBulkActionRequest.getPaymentMethod());
+      feeFineAction.setNotify(defaultBulkActionRequest.getNotifyPatron());
+      feeFineAction.setTransactionInformation(defaultBulkActionRequest.getTransactionInfo());
```

Comment:
We can also do it with a `.with()` chain to keep the same style as above.

Reply:
No we can not. There will be runtime errors in case of bulk cancel request, which is not instance of DefaultBulkActionRequest

Reply:
I meant we can put a new `.with()` chain inside this `if` instead of `set`s, just for these three fields. It just looks a little bit cleaner and uniform with what's above.

Reply:
I don't think it will be better because it will look like this: 
`feeFineAction = feeFineAction.withPaymentMethod(defaultBulkActionRequest.getPaymentMethod());`

besides we'll have to remove final modifier.

Reply:
I'll override this method.